src/client/ClientDataManager.js

```diff
@@ -73,6 +74,23 @@ class ClientDataManager {
     return null;
   }
 
+  newEmoji(data, guild) {
+    const already = guild.emojis.has(data.id);
+    if (data && !already) {
+      let emoji = new Emoji(guild, data);
+      this.client.emit(Constants.Events.GUILD_EMOJI_CREATE, guild, emoji);
+      guild.emojis.set(emoji.id, emoji);
+      return emoji;
+    }
+
+    return null;
+  }
+
+  killEmoji(emoji) {
```

Comment:
could you just make sure that killEmoji ensures that the emoji is actually an Emoji and has a valid Guild before emitting?